src/App.tsx

```diff
@@ -16,60 +16,97 @@ const goodsFromServer = [
 ];
 
 enum SortType {
-  NONE,
-  ALPABET,
-  LENGTH,
+  None = 'none',
+  Alpabet = 'alpabet',
+  Length = 'length',
+  Reverse = 'reverse',
 }
 
-// Use this function in the render method
-function getReorderedGoods(
-  goods: string[],
-  sortType: SortType,
-  isReversed: boolean,
-) {
-  // Not to mutate the original array
-  const visibleGoods = [...goods];
+export const App: FC = () => {
+  const [isStarted, setIsStarted] = useState(false);
+  const [isReversed, setIsReversed] = useState(false);
+  const [sortBy, setSortBy] = useState(SortType.None);
 
-  // Sort and reverse goods if needed
-  // ...
+  const visibleGoods = [...goodsFromServer];
 
-  return visibleGoods;
-}
-
-// DON'T save goods to the state
-type State = {
-  isStarted: boolean,
-  isReversed: boolean,
-  sortType: SortType,
-};
+  visibleGoods.sort((firstGood, secondGood) => {
+    switch (sortBy) {
+      case SortType.Alpabet:
+        return firstGood.localeCompare(secondGood);
+      case SortType.Length:
+        return firstGood.length - secondGood.length;
+      default:
+        return 0;
+    }
+  });
```

Comment:
I would suggest one more option you could have used: just create handlers and pass them to buttons, and when those button are clicked – perform the action (i.e. sort, reset, show – and so on)

why: every time your component gets re-rendered – you perform this action, is this okay?